src/hooks/useSecurityProfileAssignmentRows.ts

```diff
@@ -0,0 +1,317 @@
+          const uniqueGroups = uniqBy(updatedGroups, (g) => g.userGroup.ID + g.supplierID)
+          inheritedAssignedParties.supplierUserGroups = uniqueGroups
+        } else {
+          const additionalGroups = relatedData.adminUserGroups.filter((g) => g.ID === assignment.UserGroupID)
+          const updatedGroups = [...(inheritedAssignedParties.adminUserGroups || []), ...additionalGroups]
+          const uniqueGroups = uniqBy(updatedGroups, (group) => group.ID)
+          inheritedAssignedParties.adminUserGroups = uniqueGroups
+        }
+      }
+
+      return inheritedAssignedParties
+    },
+    [commerceRole, hasAssignmentAtLevel]
+  )
+
+  useEffect(() => {
+    const initialize = async () => {
+      const [securityProfiles, buyerUserGroups, supplierUserGroups, adminUserGroups, buyers, suppliers] =
+        await Promise.all([
+          getSecurityProfiles(),
+          getBuyerUserGroups(),
+          getSupplierUserGroups(),
+          getAdminUserGroups(),
+          getBuyers(),
+          getSuppliers()
+        ])
+
+      const relatedData: RelatedData = {
+        securityProfiles,
+        buyerUserGroups,
+        supplierUserGroups,
+        adminUserGroups,
+        buyers,
+        suppliers
+      }
+      const _rows: RowData[] = relatedData.securityProfiles.map((securityProfile) => {
+        return {
+          securityProfile: securityProfile,
+          assignments: [],
+          isInherited: false,
+          isAssigned: false,
+          inheritedAssignedParties: {}
+        }
+      })
+      assignments.forEach((assignment) => {
+        const securityProfile = relatedData.securityProfiles.find(
+          (profile) => profile.ID === assignment.SecurityProfileID
+        )
+        const isInherited = !hasAssignmentAtLevel(assignment, assignmentLevel)
+        const row = _rows.find(
+          (row) => row.securityProfile.ID === assignment.SecurityProfileID && row.isInherited === isInherited
+        )
+        if (row) {
+          row.assignments.push(assignment)
+          row.isAssigned = true
+          row.inheritedAssignedParties = updateInheritedAssignedParties(
+            assignment,
+            row.inheritedAssignedParties,
+            relatedData
+          )
+        } else {
+          _rows.push({
+            securityProfile: securityProfile,
+            assignments: [assignment],
+            isInherited: isInherited,
+            isAssigned: true,
+            inheritedAssignedParties: updateInheritedAssignedParties(assignment, {}, relatedData)
+          })
+        }
+      })
+      setRows(sortBy(_rows, (row) => row.securityProfile.Name))
+    }
+    initialize()
+  }, [
+    getAdminUserGroups,
+    getBuyerUserGroups,
+    getBuyers,
+    getSuppliers,
+    getSecurityProfiles,
+    getSupplierUserGroups,
+    hasAssignmentAtLevel,
+    assignmentLevel,
+    assignments,
+    updateInheritedAssignedParties
+  ])
+
+  return [rows]
```

Comment:
Return an object here so that in the future if we add more items people can only pull off what they need by name, instead of based on where they are at in the returned array.